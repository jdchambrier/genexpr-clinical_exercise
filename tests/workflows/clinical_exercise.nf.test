nextflow_workflow {

    name "Test Workflow CLINICALEXERCISE"
    script "workflows/clinical_exercise.nf"
    workflow "CLINICALEXERCISE"

    test("Should run without failures") {

        when {
            params {
                
                input = "${projectDir}/test_data/test_data.csv"
                managementrmd = "${projectDir}/bin/management.Rmd"
            }
        }

        then {
            assert workflow.success

            with(workflow.out) {
                //  nf-test for data management
                assert clean_data.size() == 1
                assert snapshot(clean_data).match("clean_data")
            }
        }

    }

}
