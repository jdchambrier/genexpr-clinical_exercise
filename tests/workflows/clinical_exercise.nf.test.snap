{
    "clean_data": {
        "content": [
            [
                "clean_clinical.csv:md5,6c5171a62cf177a7701abcda8d8e01a4"
            ]
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "23.10.1"
        },
        "timestamp": "2025-07-26T19:03:14.529347"
    }
}